systemd_ctypes/bus.py

```diff
@@ -220,3 +223,125 @@ def add_match(self, rule, handler):
         slot = Slot(handler)
         super().add_match(byref(slot), rule, slot.callback, slot.userdata)
         return slot
+
+    def add_object(self, path, obj):
+        obj.prepare()
+        slot = Slot(obj.message_received)
+        super().add_object(byref(slot), path, slot.callback, slot.userdata)
+        return slot
+
+
+class BaseObject:
+    @classmethod
+    def prepare(cls):
+        raise NotImplementedError
+
+    def message_received(self, message):
+        try:
+            reply = message.new_method_return()
+            handled = self.handle_method_call(message, reply)
+        except BusError:
+            assert False  # needs proper exception handling
```

Comment:
Let's add a few bad calls to the test? Nonexisting method, and method with mismatching signature?

Reply:
Sure.  I tested those manually, fwiw, and they did the right thing.  More tests welcome, of course :) :)